Docs: remove markdown docs in favor of system table generated docs for NumericIndexedVector functions

Blargian opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
Removes markdown docs in favor of system table generated docs for NumericIndexedVector functions following https://github.com/ClickHouse/clickhouse-docs/pull/4521
### Changelog category (leave one):
- Documentation (changelog entry is not required)


### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

Blargian committed:
Remove markdown docs in favor of system table generated docs

rienath reviewed (changes requested):
## 1
Some texts in documentation became redundant after the switch to autogenerated doc that includes the type information. For example,

Before:
<img width="511" height="278" alt="image" src="https://github.com/user-attachments/assets/357a7128-e408-4009-85dd-0e98dd72caab" />

After:
<img width="593" height="321" alt="image" src="https://github.com/user-attachments/assets/4f1eb764-627f-4862-a691-9f54316998fa" />

Now, it is obvious what the type is, so description `A numericIndexedVector object` is no longer needed and we can remove it. I am not sure the documentation tool allows empty entries. If not, we can instead say `The numericIndexedVector to count unique indexes from`. But I think it would be better to delete the entry as the purpose of the argument is self-evident.

This comment applies to the following sections:
* `numericIndexedVectorToMap`
* `numericIndexedVectorCardinality`
* `numericIndexedVectorAllValueSum`
* `numericIndexedVectorGetValue` (only applies to `numericIndexedVector` argument)

## 2
Maybe just "The sum..."?

<img width="610" height="415" alt="image" src="https://github.com/user-attachments/assets/7f6ac6d3-2250-47ef-9ec2-f19142d7b829" />

## 3
In 
* `numericIndexedVectorPointwiseAdd`
* `numericIndexedVectorPointwiseNotEqual`

it would be nice to change arguments section to that the argument names match the **syntax** section (i.e. numericIndexedVector → v1)

<img width="1134" height="354" alt="image" src="https://github.com/user-attachments/assets/3263bc46-9f41-4d25-885a-f7d2eaa67963" />

## 4
With the following:
* `numericIndexedVectorPointwiseDivide`
* `numericIndexedVectorPointwiseEqual`
* `numericIndexedVectorPointwiseLess`
* `numericIndexedVectorPointwiseSubtract`

we need to add that argument `v2` can also be `numericIndexedVector` (like in `numericIndexedVectorPointwiseAdd`).

## 5
With the following:

* `numericIndexedVectorPointwiseAdd`
* `numericIndexedVectorPointwiseDivide`
* `numericIndexedVectorPointwiseMultiply`
* `numericIndexedVectorPointwiseSubtract`

we can now remove `The function returns a new numericIndexedVector` from the description

Blargian committed:
incorporate review feedback

Blargian commented:
Thanks @rienath, pushed those changes. On point 1 I saw there were a few more like that. I've removed the description where it's obvious from the return type. Also renamed the parameter to `v` rather than `numericIndexedVector` in a few places for consistency.

rienath reviewed (approved):
Looks great!
